### PR TITLE
Move the ConvertTupleToStruct tests to the sdk test harness

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/46291

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@sharwell this is ready for review.

**Review by Youssef1313 — CHANGES_REQUESTED** [review]
The suppression in the codefix should probably be removed and add `[Obsolete]` attribute.

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs [hunk 1]
```diff
@@ -2,36 +2,60 @@
...
+using Microsoft.CodeAnalysis.Testing;
 using Roslyn.Test.Utilities;
 using Xunit;
 
 namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.ConvertTupleToStruct
 {
-    public class ConvertTupleToStructTests : AbstractCSharpCodeActionTest
+    using VerifyCS = CSharpCodeRefactoringVerifier<CSharpConvertTupleToStructCodeRefactoringProvider>;
+
+    [UseExportProvider]
```
Comment (Youssef1313) [inline src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:25]: Why is this needed? I don't recall any of the tests do this

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs [hunk 2]
```diff
@@ -2,36 +2,60 @@
...
+            OptionsCollection? options = null,
+            TestHost testHost = TestHost.InProcess,
+            string[]? actions = null)
+        {
+            var test = new VerifyCS.Test
+            {
+                TestCode = text,
+                FixedCode = expected,
+                TestHost = testHost,
+                CodeActionIndex = index,
```
Comment (Youssef1313) [inline src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:51]: I'd use both CodeActionIndex and CodeActionEquivalenceKey

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:25]: @sharwell i needed this because when i did the OOP composition i got the errro about needing to say that we UseExportProvider.  Can you advise on if this is needed?

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:51]: these are refactorign.  so as per our default, we don't set equivalence keys (since we dont' have any scenarios we expect to use them).

**Comment by CyrusNajmabadi** [comment]
> The suppression in the refactoring provider should probably be removed and add [Obsolete] attribute:

Good catch.  Thanks!

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs [hunk 3]
```diff
@@ -870,7 +900,32 @@ public static implicit operator (object a, object b)(NewStruct value)
...
+                TestCode = text,
+                FixedCode = expected,
+                TestHost = host,
+                ExpectedDiagnostics =
+                {
+    // /0/Test0.cs(6,22): error CS0103: The name 'Goo' does not exist in the current context
+    DiagnosticResult.CompilerError("CS0103").WithSpan(7, 22, 7, 25).WithArguments("Goo"),
+    // /0/Test0.cs(6,32): error CS0103: The name 'Bar' does not exist in the current context
+    DiagnosticResult.CompilerError("CS0103").WithSpan(7, 32, 7, 35).WithArguments("Bar"),
+                },
```
Comment (Youssef1313) [inline src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:914]: I'd do one of the following:

- Get rid of the diagnostics from the test.
- Use the `{|CS0103:code|}` syntax.
- Use `{|#0:code|}` syntax and use `WithLocation(0)` instead of `WithSpan`.

Will let you and @sharwell decide the appropriate change if any.

Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:914]: > Get rid of the diagnostics from the test.

in this case, the tewst is intentionally testint out what we do with incorrect code.  so i want to keep that :)

> Use the {|CS0103:code|} syntax.

I'm fine with that.  but i'm also fine with how this looks (which is also how the test harness outputs the diags).  i'm honestly not wanting to spend much effort here.

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs [hunk 4]
```diff
@@ -1809,7 +1887,82 @@ public static implicit operator (int a, int a)(NewStruct value)
...
+            {
+                TestCode = text,
+                FixedCode = expected,
+                TestHost = host,
+                ExpectedDiagnostics =
+                {
+                    // /0/Test0.cs(6,25): error CS8127: Tuple element names must be unique.
+                    DiagnosticResult.CompilerError("CS8127").WithSpan(6, 25, 6, 26),
+                },
+                FixedState =
```
Comment (Youssef1313) [inline src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:1900]: Looks like the fixed state has much more diagnostics than the test state. I haven't looked at what the refactoring did yet, but it's worth giving a look to see if this high number of diagnostics indicate a bug in the refactoring.

File: src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs [hunk 5]
```diff
@@ -85,8 +92,70 @@ public Test()
...
 #if !CODE_STYLE
+            private readonly List<AdhocWorkspace> _workspaces = new();
+
             protected override AnalyzerOptions GetAnalyzerOptions(Project project)
                 => new WorkspaceAnalyzerOptions(base.GetAnalyzerOptions(project), project.Solution);
+
+            /// <summary>
+            /// The <see cref="TestHost"/> we want this test to run in.  Defaults to <see cref="TestHost.InProcess"/> if unspecified.
+            /// </summary>
+            public TestHost TestHost { get; set; } = TestHost.InProcess;
```
Comment (Youssef1313) [inline src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs:103]: Not sure if it's worth testing both in process and out-of-proc.

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs (hunk 4 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:1900]: yup.  it's somethign we could consider changing how we emit.  but this test accurately describes how we are emitting today.  in this case, the large number is due to the fact that the tuple has identical member names.  when that becomes a struct, you end up with a ton of errors about that.

File: src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs (hunk 5 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs:103]: it is :)   it catches real issues.

Comment (CyrusNajmabadi) [reply src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs:103]: (in this case, the feature has a large oop piece of work that it does, so we really want to validate that that is working properly).

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs (hunk 4 above)
Comment (Youssef1313) [reply src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:1900]: Looked on this now. This was due to duplicate tuple element names. The generated struct had duplicated the name as well, so it had that much of diagnostics. It's probably fine given the original code is problematic.

Comment (Youssef1313) [reply src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:1900]: > yup. it's somethign we could consider changing how we emit. but this test accurately describes how we are emitting today. in this case, the large number is due to the fact that the tuple has identical member names. when that becomes a struct, you end up with a ton of errors about that.

Yeah I think it's fine. While things could be improved, it's not necessary and also completely of out scope for this PR.

File: src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs (hunk 5 above)
Comment (Youssef1313) [reply src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs:103]: @CyrusNajmabadi Thanks for clarifying

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs [hunk 6]
```diff
@@ -50,7 +74,7 @@ class Test
 {
     void Method()
     {
-        var t1 = new {|Rename:NewStruct|}(a: 1, b: 2);
```
Comment (Youssef1313) [inline src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:53]: It looks like `{|Rename: |}` was doing extra validation that's now lost?

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs [hunk 7]
```diff
@@ -804,13 +832,14 @@ public static implicit operator (int a, int b)(NewStruct value)
...
 ";
-            await TestInRegularAndScriptAsync(text, expected, options: this.PreferImplicitTypeWithInfo(), testHost: host);
+            await Test(text, expected, options: PreferImplicitTypeWithInfo(), testHost: host);
         }
 
         [Theory, CombinatorialData, Trait(Traits.Feature, Traits.Features.CodeActionsConvertTupleToStruct)]
         public async Task TestNonLiteralNames(TestHost host)
         {
             var text = @"
+using System.Collections.Generic;
```
Comment (Youssef1313) [inline src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:842]: This is out-of-scope for this PR. But I think this `using` should be in the `expected` state even if it doesn't in the original input (i.e, have AddImportsAnnotation on the node for `EqualityComparer`). I personally think the input should remain as-is for this PR, and the behavioral change is addressed in a followup.

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs [hunk 8]
```diff
@@ -3115,11 +3278,32 @@ void Goo()
...
+                    Sources = { expected1 },
+                    AdditionalProjects =
+                    {
+                        ["DependencyProject"] = { Sources = { expected2 } }
+                    },
+                },
+            };
+
+            test.Options.AddRange(PreferImplicitTypeWithInfo());
+            await test.RunAsync();
```
Comment (Youssef1313) [inline src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:3306]: Hard to tell what the action with index 3 means? What are the other offered actions?

Could this test use `ExactActionSetOffered` (or any other form that clarifies it a bit)?

File: src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs [hunk 9]
```diff
@@ -85,8 +92,70 @@ public Test()
...
+            /// </summary>
+            public string[]? ExactActionSetOffered { get; set; }
+
+            protected override ImmutableArray<CodeAction> FilterCodeActions(ImmutableArray<CodeAction> actions)
+            {
+                var result = base.FilterCodeActions(actions);
+
+                if (ExactActionSetOffered != null)
+                {
+                    Verify.Equal(ExactActionSetOffered.Length, result.Length, "Different number of code actions provided");
```
Comment (Youssef1313) [inline src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs:117]: nit:

```suggestion
                    Verify.Equal(ExactActionSetOffered.Length, result.Length, $"Different number of code actions provided. Expected: {ExactActionSetOffered.Length}, Actual: {result.Length}.");
```

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs (hunk 6 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:53]: Yes.  It's a known limitation.  @sharwell  is tracking an issue to bring this back at some point.  When we do, we'd update tehse tests.

File: src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs [hunk 10]
```diff
@@ -85,8 +92,70 @@ public Test()
...
+            /// </summary>
+            public TestHost TestHost { get; set; } = TestHost.InProcess;
+
+            /// <summary>
+            /// The set of code action <see cref="CodeAction.Title"/>s offered the user in this exact order.
+            /// Set this to ensure that a very specific set of actions is offered (or not).
+            /// </summary>
+            public string[]? ExactActionSetOffered { get; set; }
+
+            protected override ImmutableArray<CodeAction> FilterCodeActions(ImmutableArray<CodeAction> actions)
```
Comment (Youssef1313) [inline src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs:111]: This method is under `#if !CODE_STYLE`. Was that intended?

File: src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs (hunk 9 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs:117]: i think that isn't really needed as that's what Verify.Equal should already indicate :)

File: src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs (hunk 10 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs:111]: Yes.  for now, i'm only modifying the testing abilities outside of the CODE_STYLE layer.  @sharwell  to advise if we'd want this to be for both layers.  I'm deferring to him on all design decisions for this test harness.

File: src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs (hunk 9 above)
Comment (Youssef1313) [reply src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs:117]: > i think that isn't really needed as that's what Verify.Equal should already indicate :)

Yeah. I think I should sleep :smile:

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs (hunk 1 above)
Comment (sharwell) [reply src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:25]: Yes all the code action tests will need this

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs (hunk 3 above)
Comment (sharwell) [reply src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:914]: We use explicit `ExpectedDiagnostics` for compiler errors that are likely the result of an incorrectly written test, and `{|CS0103:...|}` for cases where the error is intentionally part of the test input/output.

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs (hunk 8 above)
Comment (sharwell) [reply src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:3306]: The recommended approach is always specifying _both_ `CodeActionIndex` and `CodeActionEquivalenceKey`. This form asserts that the action presented at a given index has a given equivalence key, which is both easy to read and strong assurance of intended order.

File: src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs (hunk 9 above)
Comment (sharwell) [reply src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs:117]: 💡 Use `SequenceEqual` with a comparer.

File: src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs (hunk 10 above)
Comment (sharwell) [reply src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs:111]: This one should work in both. `CreateWorkspace` is restricted because the OOP compositions are not available in code style layer testing.

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs (hunk 3 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:914]: ah ok.  will do.

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs (hunk 8 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:3306]: ok.  can do so.

File: src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs (hunk 9 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs:117]: will do.

File: src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs (hunk 10 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs:111]: sgtm.

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs (hunk 7 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:843]: yes, that's a good point.  i would ok with us making such a behavior change in teh future :)

**Comment by CyrusNajmabadi** [comment]
Ok.  Feedback responded to.

File: src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs [hunk 11]
```diff
@@ -84,9 +91,70 @@ public Test()
...
+            {
+                var result = base.FilterCodeActions(actions);
+
+                if (ExactActionSetOffered != null)
+                {
+                    Verify.Equal(ExactActionSetOffered.Length, result.Length, "Different number of code actions provided");
+                    Verify.SequenceEqual(ExactActionSetOffered, result.SelectAsArray(a => a.Title));
```
Comment (sharwell) [inline src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs:107]: The error message in this case won't be helpful. Only `SequenceEqual` is needed.

```suggestion
                    Verify.SequenceEqual(ExactActionSetOffered, result.SelectAsArray(a => a.Title));
```

File: src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs [hunk 12]
```diff
@@ -84,9 +91,70 @@ public Test()
 
             public string? EditorConfig { get; set; }
 
+            /// <summary>
+            /// The set of code action <see cref="CodeAction.Title"/>s offered the user in this exact order.
+            /// Set this to ensure that a very specific set of actions is offered (or not).
```
Comment (sharwell) [inline src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs:96]:
```suggestion
            /// Set this to ensure that a very specific set of actions is offered.
```

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs [hunk 13]
```diff
@@ -2,36 +2,61 @@
...
+        private static OptionsCollection PreferImplicitTypeWithInfo()
+            => new OptionsCollection(LanguageNames.CSharp)
+            {
+                { CSharpCodeStyleOptions.VarElsewhere, true, NotificationOption2.Suggestion },
+                { CSharpCodeStyleOptions.VarWhenTypeIsApparent, true, NotificationOption2.Suggestion },
+                { CSharpCodeStyleOptions.VarForBuiltInTypes, true, NotificationOption2.Suggestion },
+            };
 
-        #region update containing member tests
+        private static async Task Test(
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:38]:
```suggestion
        private static async Task TestAsync(
```

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs [hunk 14]
```diff
@@ -804,27 +833,29 @@ public static implicit operator (int a, int b)(NewStruct value)
...
 ";
-            await TestInRegularAndScriptAsync(text, expected, options: this.PreferImplicitTypeWithInfo(), testHost: host);
+            await Test(text, expected, options: PreferImplicitTypeWithInfo(), testHost: host);
         }
 
         [Theory, CombinatorialData, Trait(Traits.Feature, Traits.Features.CodeActionsConvertTupleToStruct)]
         public async Task TestNonLiteralNames(TestHost host)
         {
             var text = @"
+using System.Collections.Generic;
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:843]: 📝 These `using` directives should not be added as part of this PR. The change hides a design limitation in the refactoring.

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs [hunk 15]
```diff
@@ -2865,11 +2969,25 @@ void Goo()
...
-        </Document>
-    </Project>
-</Workspace>";
-            await TestInRegularAndScriptAsync(text, expected, index: 2, options: this.PreferImplicitTypeWithInfo(), testHost: host);
+}";
+
+            var test = new VerifyCS.Test
+            {
+                CodeActionIndex = 2,
+                CodeActionEquivalenceKey = null,
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:2977]: 📝 All tests that set `CodeActionIndex` need to also set a non-null `CodeActionEquivalenceKey`. You may need to fix the refactoring implementation to set this.

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs (hunk 2 above)
Comment (sharwell) [reply src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:51]: Marking this one as active. When testing code actions, the `CodeActionIndex` is optional but the `CodeActionEquivalenceKey` is required (and cannot be null). Generally both are used together. This approach substantially improves readability of the tests, and hardens the tests against clarity issues that can occur if the ordering changes. The implementation may need to be updated to meet this requirement.

**Review by sharwell — CHANGES_REQUESTED** [review]
Primary requested change is the test behavior change involving `using` directives.

Secondary requested changes is assigning equivalence keys to the code actions to make this refactoring compatible with the new test library.

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:51]: k, will do.

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs (hunk 15 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs]: have done this now.

File: src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs (hunk 14 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs:848]: have split these into tests that do both forms.

**Review by sharwell — APPROVED** [review]
Approved with a suggestion
